File: CRM.Api/Models/CustomFields/CustomField.cs
namespace CRM.Api.Models.CustomFields
{
    public class CustomField
    {
        public int Id { get; set; }
        public string EntityType { get; set; } = string.Empty; // Contact, Company, Opportunity
        public string FieldName { get; set; } = string.Empty;
        public string DisplayName { get; set; } = string.Empty;
        public CustomFieldType FieldType { get; set; }
        public bool IsRequired { get; set; }
        public bool IsActive { get; set; } = true;
        public string? DefaultValue { get; set; }
        public string? ValidationRules { get; set; } // JSON
        public string? Options { get; set; } // JSON for dropdown options
        public int DisplayOrder { get; set; }
        public string? SectionName { get; set; }
        public string? HelpText { get; set; }
        public int CreatedBy { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }

        // Navigation
        public ICollection<CustomFieldValue> Values { get; set; } = new List<CustomFieldValue>();
    }

    public enum CustomFieldType
    {
        Text = 0,
        Textarea = 1,
        Number = 2,
        Decimal = 3,
        Date = 4,
        DateTime = 5,
        Dropdown = 6,
        MultiSelect = 7,
        Checkbox = 8,
        URL = 9,
        Email = 10,
        Phone = 11,
        Currency = 12,
        Percentage = 13
    }
}

File: CRM.Web/src/types/CustomField.ts
export type CustomFieldType =
    | 'Text'
    | 'Number'
    | 'Date'
    | 'Boolean'
    | 'Select'
    | 'MultiSelect'
    | 'TextArea'
    | 'Decimal'
    | 'DateTime'
    | 'Checkbox'
    | 'URL'
    | 'Email'
    | 'Phone'
    | 'Currency'
    | 'Percentage';

export interface ValidationRule {
    ruleType: string;
    value: string;
    errorMessage: string;
}

export interface FieldOption {
    label: string;
    value: string;
    isDefault: boolean;
}

export interface CustomField {
    id: number;
    entityType: string;
    fieldName: string;
    displayName: string;
    fieldType: number; // 0=Text, 1=Textarea, 2=Number, 3=Decimal, 4=Date, 5=DateTime, 6=Dropdown, 7=MultiSelect, 8=Checkbox, 9=URL, 10=Email, 11=Phone, 12=Currency, 13=Percentage
    isRequired: boolean;
    isActive: boolean;
    defaultValue?: string;
    validationRules?: ValidationRule[];
    options?: FieldOption[];
    displayOrder: number;
    sectionName?: string;
    helpText?: string;
    fieldKey?: string;
    optionsJson?: string;
}

export interface CreateCustomFieldDto {
    entityType: string;
    fieldName: string;
    displayName: string;
    fieldType: string;
    isRequired: boolean;
    defaultValue?: string;
    validationRules?: ValidationRule[];
    options?: FieldOption[];
    sectionName?: string;
    helpText?: string;
}

export interface UpdateCustomFieldDto {
    displayName: string;
    isRequired: boolean;
    isActive: boolean;
    defaultValue?: string;
    validationRules?: ValidationRule[];
    options?: FieldOption[];
    displayOrder: number;
    sectionName?: string;
    helpText?: string;
}

export interface CustomFieldValue {
    customFieldId: number;
    fieldName: string;
    displayName: string;
    fieldType: string;
    value: any;
}

export interface SaveCustomFieldValuesDto {
    values: Record<string, any>;
}
